lib/src/component_declaration/component_type_checking.dart

```diff
@@ -32,6 +32,96 @@ import 'package:react/react_client/react_interop.dart';
 //   Component type registration and internal type metadata management
 // ----------------------------------------------------------------------
 
+/// Extensions to allow setting type-checking-related metadata on components.
+extension UiFactoryTypeMeta on UiFactory {
+  /// Sets metadata for the underlying component class/type associated with this
+  /// factory, functionally equivalent to the [annotations.Component2] annotation.
+  ///
+  /// However, unlike the annotation, this can be use for function components.
+  ///
+  /// ## subtypeOf arguments: [subtypeOfFactory]/[subtypeOfRaw]
+  ///
+  /// The raw JS component type that is this component's "parent type".
+  ///
+  /// Used to enable inheritance in component type-checking in `isComponentOfType`.
```

Comment:
#nit, should this be wrapped in braces instead?
```suggestion
  /// Used to enable inheritance in component type-checking in [isComponentOfType].
```